### PR TITLE
ISSUE-135 Add a phase to validate Cassandra + RabbitMQ + Swift + Ldap product

### PR DESCRIPTION
**Pull request by hoangdat**

### CODE REVIEW

**Comment by hoangdat** [comment]
@rouazana @chibenwa @trantienduchn Please review for me

**Review by trantienduchn — APPROVED** [review]

**Review by chibenwa — COMMENTED** [review]
How about publish docker images to dockerhub?

File: workflow-job [hunk 1]
```diff
@@ -423,6 +440,38 @@ flows["${sha1}"] = {
...
             testingJava8.success();
 
+            //Deploying Guice Server with Cassandra + RabbitMQ + Swift + Ldap
+            sh "cd dockerfiles/run/guice/cassandra-rabbitmq-ldap; docker cp ${containers.keystore}:${keystorePath} destination/conf"
+            sh "cd dockerfiles/run/guice/cassandra-rabbitmq-ldap; docker cp ${containers.jamesTest}:${jarPathCassandraRabbitMQLdap} destination"
+            sh "cd dockerfiles/run/guice/cassandra-rabbitmq-ldap; docker cp ${containers.jamesTest}:${libPathCassandraRabbitMQLdap} destination"
+            sh "cd dockerfiles/run/guice/cassandra-rabbitmq-ldap; docker cp ${containers.jamesTest}:${cliJarPathCassandraRabbitMQLdap} destination"
+            sh "cd dockerfiles/run/guice/cassandra-rabbitmq-ldap; docker cp ${containers.jamesTest}:${clilibPathCassandraRabbitMQLdap} destination"
+
+            sh "docker run --detach=true --name=${containers.cassandra} ${images.cassandra}"
```
Comment (chibenwa) [inline workflow-job]: There will be conflicts as these containers are not removed before subsequent docker runs

Comment (hoangdat) [reply workflow-job]: IMO, all containers and images will be deleted in the `finally` block of code.

**Comment by hoangdat** [comment]
> How about publish docker images to dockerhub?

I don't know how to evaluate it. Please guide me more details.

File: workflow-job (hunk 1 above)
Comment (chibenwa) [reply workflow-job]: Yes but the next step is creating them again - we need to remove them - to reset data - before plating other deployment tests

File: workflow-job [hunk 2]
```diff
@@ -116,10 +116,12 @@ public class Images implements Serializable {
     def elasticSearch = 'elasticsearch:2.4.6'
     def rabbitmq = "rabbitmq:3.7.7-management"
     def swift = "linagora/openstack-keystone-swift:pike"
+    def ldap = "linagora/james-ldap-project:latest"
```
Comment (chibenwa) [inline workflow-job]: This is not a LDAP image but a James image.

More generally we made the choice before to not do deployment tests for LDAP images - just dockerize + push and it prooved to not work too bad so far. I suggest that we could be dropping the "deployment validation part" here that you are writing and that is not mentioned in the ticket.

**Review by chibenwa — APPROVED** [review]

File: workflow-job [hunk 3]
```diff
@@ -597,6 +617,9 @@ flows["${sha1}"] = {
             def tagName = tag();
             sh "docker login -u ${dUser} -p ${dPassword}"
 
+            sh "docker tag ${images.jamesCassandraRabbitMQLdap} linagora/james-rabbitmq-ldap-project:${tagName}"
```
Comment (trantienduchn) [inline workflow-job]: by existing convention, I prefer to have a `linagora/james-cassandra-rabbitmq-ldap-project` image

**Review by mbaechler — APPROVED** [review]

**Review by trantienduchn — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**